core/src/main/java/org/mule/runtime/core/internal/processor/strategy/ProactorStreamEmitterProcessingStrategyFactory.java

```diff
@@ -128,12 +130,13 @@ public ProactorStreamEmitterProcessingStrategy(Supplier<Scheduler> ringBufferSch
     @Override
     public Sink createSink(FlowConstruct flowConstruct, ReactiveProcessor function) {
       final long shutdownTimeout = flowConstruct.getMuleContext().getConfiguration().getShutdownTimeout();
-      List<ReactorSink<CoreEvent>> sinks = new ArrayList<>();
-      int concurrency = maxConcurrency < subscribers ? maxConcurrency : subscribers;
+      final int sinkBufferSize = bufferSize / sinksCount;
       reactor.core.scheduler.Scheduler scheduler = fromExecutorService(decorateScheduler(getCpuLightScheduler()));
-      for (int i = 0; i < concurrency; i++) {
+      List<ReactorSink<CoreEvent>> sinks = new ArrayList<>();
+
+      for (int i = 0; i < sinksCount; i++) {
         Latch completionLatch = new Latch();
-        EmitterProcessor<CoreEvent> processor = EmitterProcessor.create(bufferSize / concurrency);
+        EmitterProcessor<CoreEvent> processor = EmitterProcessor.create(sinkBufferSize);
         processor.publishOn(scheduler).doOnSubscribe(subscription -> currentThread().setContextClassLoader(executionClassloader))
```

Comment:
can this `publishOn` be moved to `onPipeline`?

Reply:
if this is moved, also the `doOnSubscribe` goes with it

Reply:
No, with that change the processing strategy doesn't even work (fails every test)

Reply:
Fixed